### PR TITLE
Wait for server process in client_server_test

### PR DESCRIPTION
**Pull request by cocreature**
withCreateProcess kills the server process but it doesn’t wait for it
to finish. The `typed-process` version does things properly here which
is also why we switched to it in other places.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: bazel_tools/client_server/runner_with_port_file/Main.hs [hunk 1]
```diff
@@ -28,7 +28,7 @@ main = do
     let portFile = tempDir </> "portfile"
     let interpolatedServerArgs = map (replace "%PORT_FILE%" portFile) splitServerArgs
     let serverProc = proc serverExe interpolatedServerArgs
-    withCreateProcess serverProc $ \_stdin _stdout _stderr ph -> do
-      port <- readPortFile ph maxRetries portFile
+    withProcessTerm serverProc $ \ph -> do
```
Comment (sofiafaro-da) [inline bazel_tools/client_server/runner_with_port_file/Main.hs:31]: I'm confused. If you want to wait for it to finish, shouldn't you use `withProcessWait` here?

**Review by sofiafaro-da — APPROVED** [review]
Thanks!

File: bazel_tools/client_server/runner_with_port_file/Main.hs (hunk 1 above)
Comment (cocreature) [reply bazel_tools/client_server/runner_with_port_file/Main.hs:31]: I want to kill it gracefully by sending SIGTERM and then wait for it to finish. That’s what `withProcessTerm` does. `withProcessWait` on the other hand does not send SIGTERM and only waits for it to finish which for something like sandbox will never happen on its own.

Comment (sofiafaro-da) [reply bazel_tools/client_server/runner_with_port_file/Main.hs:31]: Ah, understood. Thanks for explaining!

**Pull request merged**